provider/resources.go

```diff
@@ -37,26 +39,10 @@ const (
 	mainMod = "index"
 )
 
-func makeMember(mod string, mem string) tokens.ModuleMember {
-	return tokens.ModuleMember(mainPkg + ":" + mod + ":" + mem)
-}
-
-func makeType(mod string, typ string) tokens.Type {
-	return tokens.Type(makeMember(mod, typ))
-}
-
-func makeDataSource(mod string, res string) tokens.ModuleMember {
-	return makeMember(mod, res)
-}
-
-func makeResource(mod string, res string) tokens.Type {
-	fn := string(unicode.ToLower(rune(res[0]))) + res[1:]
-	return makeType(mod+"/"+fn, res)
-}
+func ref[T any](t T) *T { return &t }
 
-func tfLicenseTypeRef(license tfbridge.TFProviderLicense) *tfbridge.TFProviderLicense {
```

Comment:
Can we add a blurb as to why we're removing the license?

Reply:
We're not removing the license. 

I've generalized this function. All this function does is take a tfbridge.TFProviderLicense and grab a reference to it. This is necessary since `tfbridge.MITLicenseType` is a constant, and go makes it illegal to take references to literals (constants count as literals everywhere in go).

The new function `ref` (L42) does the same thing. We still apply the license on L61.